### PR TITLE
fix(export): #CO-1795 translate correctly categories

### PR DESCRIPTION
**Pull request by Ozen40**
## Describe your changes
When you export a ticket, it will now correctly translate category addresses to their names
## Checklist tests

- Export tickets and check if categories are correctly translate

## Issue ticket number and link
[CO-1795](https://jira.support-ent.fr/browse/CO-1795)
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/net/atos/entng/support/helpers/CSVHelper.java [hunk 1]
```diff
@@ -22,4 +28,13 @@ public static ExportFile getExportFile(String filename, String csv) {
         return new ExportFile(buffer, "text/csv; charset=utf-8", filename);
     }
 
+    public static JsonArray translateTicketCategory(UserInfos user, JsonArray ticketsResults) {
+        return new JsonArray(ticketsResults.stream().filter(JsonObject.class::isInstance)
+                .map(JsonObject.class::cast).map(ticket -> {
```
Comment (BastienBenard) [inline src/main/java/net/atos/entng/support/helpers/CSVHelper.java:33]:
```suggestion
        return new JsonArray(ticketsResults.stream()
                .filter(JsonObject.class::isInstance)
                .map(JsonObject.class::cast)
                .map(ticket -> {
```

File: src/main/java/net/atos/entng/support/helpers/UserInfosHelper.java [hunk 2]
```diff
@@ -87,4 +88,10 @@ public static List<UserInfos.Action> getUserActionsFromJSONArray(JsonArray actio
         return ((List<JsonObject>) actions.getList()).stream().map(UserInfosHelper::getUserActionFromJSON).collect(Collectors.toList());
      }
 
+    public static String getAppName(UserInfos user, String appAddress) {
+        return user.getApps().stream()
+                .filter(app -> Objects.equals(app.getAddress(), appAddress))
+                .map(UserInfos.Application::getDisplayName).findFirst().orElse("");
```
Comment (BastienBenard) [inline src/main/java/net/atos/entng/support/helpers/UserInfosHelper.java:94]: Une instruction par ligne
```suggestion
        return user.getApps().stream()
                .filter(app -> Objects.equals(app.getAddress(), appAddress))
                .map(UserInfos.Application::getDisplayName)
                .findFirst()
                .orElse("");
```

File: src/main/java/net/atos/entng/support/services/impl/TicketServiceSqlImpl.java [hunk 3]
```diff
@@ -911,9 +911,21 @@ public Future<JsonObject> countTickets(UserInfos user, JsonObject schoolId){
...
+			List<String> scopesList = adminLocal.getScope();
+			if (scopesList != null && !scopesList.isEmpty()) {
+				query.append(Sql.listPrepared(scopesList));
+				values.addAll(new JsonArray(scopesList));
+			}
+		} else {
+			query.append(Sql.listPrepared(user.getStructures()));
+			values.addAll(new JsonArray(user.getStructures()));
+		}
+		sql.prepared(query.toString(), values, validResultHandler(PromiseHelper.handler(promise)));
```
Comment (BastienBenard) [inline src/main/java/net/atos/entng/support/services/impl/TicketServiceSqlImpl.java:928]: Ce serait bien de faire un model. + test

File: src/main/java/net/atos/entng/support/controllers/TicketController.java [hunk 4]
```diff
@@ -981,19 +986,30 @@ public void directExport(HttpServerRequest request) {
...
                         .compose(ticketServiceSql::getTicketsFromStructureIds)
                         .onComplete(promise);
-                else ticketServiceSql.getUserTickets(user).onComplete(promise);
+                else ticketServiceSql.getUserTickets(user).onComplete(ar -> {
+                    List<TicketModel> ticketModels = ar.result();
+                    JsonArray jsonArray = new JsonArray();
+                    for (TicketModel ticketModel : ticketModels) {
+                        jsonArray.add(ticketModel.toJson());
+                    }
+                    promise.complete(jsonArray);
```
Comment (qpic-cgi) [inline src/main/java/net/atos/entng/support/controllers/TicketController.java:1011]: https://github.com/CGI-OPEN-ENT-NG/minibadge/blob/dev/src/main/java/fr/cgi/minibadge/helper/RequestHelper.java#L46

tu peux faire une méthode dans un utils, un peut similaire à ça

File: src/main/java/net/atos/entng/support/export/TicketExportWorker.java [hunk 5]
```diff
@@ -88,13 +85,17 @@ public void handle(Message<JsonObject> event) {
...
+        JsonArray tickets = new JsonArray();
+        UserInfos user = UserInfosHelper.getUserInfosFromJSON(params.getJsonObject(Ticket.USER));
         getStructureIds(structureId, params)
                 .compose(structureIds -> ticketServiceSql.getTicketsFromStructureIds(structureIds))
-                .compose(tickets -> ticketService.getProfileFromTickets(tickets, i18nConfig))
-                .compose(ticketService::getSchoolFromTickets)
+                .compose(ticketsResults -> {
+                    tickets.addAll(CSVHelper.translateTicketCategory(user, ticketsResults));
+                    return CompositeFuture.all(ticketService.getProfileFromTickets(tickets, i18nConfig),
+                            ticketService.getSchoolFromTickets(tickets));
```
Comment (qpic-cgi) [inline src/main/java/net/atos/entng/support/export/TicketExportWorker.java:95]: dupliqué du [TicketController.java](https://github.com/OPEN-ENT-NG/support/pull/58/files#diff-8ac60e719063e8ae37baca0df228c7855265316b2c7231ba680c3aed218dfa40) non ?
il n'y a pas moyen de mutualiser ?

**Review by qpic-cgi — APPROVED** [review]

File: src/main/java/net/atos/entng/support/helpers/IModelHelper.java [hunk 6]
```diff
@@ -9,6 +9,7 @@
 import io.vertx.core.json.JsonObject;
 import io.vertx.core.logging.Logger;
 import io.vertx.core.logging.LoggerFactory;
+import net.atos.entng.support.model.TicketModel;
```
Comment (BastienBenard) [inline src/main/java/net/atos/entng/support/helpers/IModelHelper.java:12]: Inutile

File: src/main/java/net/atos/entng/support/model/TicketModel.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    private String schoolId;
+
+    public TicketModel() {
+    }
+
+    public TicketModel(JsonObject jsonObject) {
+        this.id = jsonObject.getInteger(Ticket.ID);
+        this.owner = jsonObject.getString(Ticket.OWNER);
+        this.subject = jsonObject.getString(Ticket.SUBJECT);
+        this.description = jsonObject.getString("description");
```
Comment (BastienBenard) [inline src/main/java/net/atos/entng/support/model/TicketModel.java:25]: Magic string

**Review by BastienBenard — APPROVED** [review]

**Pull request merged**